docs/spec/crisis/02_messages.md

```diff
@@ -0,0 +1,27 @@
+# Messages
+
+In this section we describe the processing of the crisis messages and the
+corresponding updates to the state. 
+
+## MsgVerifyInvariant
+
+Blockchain invariants can be check using the `MsgVerifyInvariant` message. 
+
+```golang
+type MsgCreateValidator struct {
+	Sender         sdk.AccAddress 
+	InvariantRoute string
+}
+```
+
+This message is expected to fail if: 
+ - the sender does not have enough coins for the constant fee
+ - the invariant route is not registered 
+
+This message checks the invariant provided, if the invariant is broken it
+panics halting the blockchain. If the invariant is broken, the constant fee is
+refunded to the message sender from the community pool, however if the
+invariant is not broken, the constant fee will not be refunded.
+
+This message should never fail if the invariant is broken, even if there are
+insufficient funds to refund the sender of the message. 
```

Comment:
When would there be insufficient funds? Also I don't understand this sentence - if the invariant is broken, the chain should panic & halt - so why would we care if it failed?

Reply:
"insufficient funds in the community pool" - hopefully that is more clear.

> so why would we care if it failed

we don't, that's what this statement is is sayin'

Reply:
updated to `If the invariant is broken the chain will always halt - even if there are
insufficient funds in the community pool to refund the sender of the message. `

Reply:
In the end I ended up just removing this statement and commenting out the associated code as it is never used because the chain halts (will be different once circuit breaker implemented)